File: ruby/problem2.rb
#!usr/bin/env ruby
# Patrick Dean
# Project Euler: #002


def fibonacci limit
    a, b = 0, 1
    while a < limit
        yield a
        a, b = b, a + b
    end
end

a = []
fibonacci(4000000) { |x| a << x if x % 2 == 0 }

puts a.inject { |sum, x| sum + x }

File: python/problem2.py
#!/usr/bin/env python
# Patrick Dean
# Project Euler: #002
# Find the sum of all even valued Fibonnaci number less than 4,000,000.

def fibonacci(limit = 0):
    a, b = 0, 1
    while True:
        if limit and a > limit:
            break
        yield a
        a, b = b, a + b
        
        
print sum(x for x in fibonacci(4000000) if x % 2 == 0)